src/components/SelectionPills/SelectionPills.js

```diff
@@ -0,0 +1,196 @@
+    const { selectAllLabel, onSelectPill } = this.props
+    const { pillsList, allSelected } = this.state
+    const selectedPill = { id: 'selectAllPill', isSelected: !allSelected, text: selectAllLabel }
+    const newPillList = pillsList.map(p => {
+      return {
+        ...p,
+        isSelected: false,
+      }
+    })
+
+    this.setState({
+      pillsList: newPillList,
+      allSelected: !allSelected,
+    })
+    onSelectPill(e, selectedPill, newPillList)
+  }
+
+  onSelectPill = (e, pill) => {
+    const { onSelectPill, pills, parentControlledState } = this.props
+    const selectedPill = { ...pill, isSelected: !pill.isSelected }
+    if (parentControlledState) {
+      return onSelectPill(e, selectedPill, pills)
+    }
+
+    const { pillsList, selectedCount } = this.state
+    const newPillList = pillsList.map(p => {
+      if (p.id !== pill.id) {
+        return p
+      }
+      return selectedPill
+    })
+
+    this.setState({
+      pillsList: newPillList,
+      selectedCount: pill.isSelected ? selectedCount - 1 : selectedCount + 1,
+      allSelected: false,
+    })
+    onSelectPill(e, selectedPill, newPillList)
+  }
+
+  // Disable not selected pills when max pills reached. Not applicable when parent controlled
+  isDisabledPill = pill => {
+    const { maxSelectionCount, parentControlledState } = this.props
+    const { selectedCount } = this.state
+    if (parentControlledState || pill.isSelected) return false
+    if (!selectedCount || !maxSelectionCount) return false
+    return selectedCount >= maxSelectionCount
+  }
+
+  renderSelectAllPill = () => {
+    const {
+      maxSelectionCount,
+      includeSelectAll,
+      selectAllLabel,
+      parentControlledState,
+    } = this.props
+    const { allSelected } = this.state
+    if (maxSelectionCount || !includeSelectAll || parentControlledState) return
+
+    return (
+      <SelectionPill
+        onClick={e => this.onSelectAll(e)}
+        isSelected={allSelected}
+        text={selectAllLabel}
+        aria={{ label: selectAllLabel }}
+        key="selectAllPill"
+        id="selectAllPill"
+        parentControlledState
+      />
+    )
+  }
+
+  renderLabel = () => {
+    const { label, style } = this.props
+    if (!label) return null
+    return (
+      <span id="selectionPillsLabel" style={style.labelStyle}>
+        {label}
+      </span>
+    )
+  }
+
+  renderPill = (pill, idx) => (
+    <SelectionPill
+      isDisabled={this.isDisabledPill(pill)}
+      onClick={e => this.onSelectPill(e, pill)}
+      key={`selectionPill-${idx}`}
+      id={pill.id || `selectionPill-${pill.text}-${idx}`}
+      parentControlledState
+      {...pill}
+    />
+  )
+
+  render() {
+    const { listAttributes, elementAttributes, parentControlledState, style } = this.props
+    const listToRender = parentControlledState ? this.props.pills : this.state.pillsList
+    const componentStyles = getStyles({ externalStyles: style })
+
+    return (
+      <ScrollTrack>
```

Comment:
The only thoughts I have on the design are around `ScrollTrack`, is this something you're comfortable letting callers handle? e.g. I'm working on something `SelectionPills` would be great for, but mine are laid out as a grid vs. a scrollable list.